Bug: make mac-app not compatible with m1 chip

## Steps to reproduce the behavior

Run the the command `make mac-app` using a MacBook Pro - Apple M1 Chip . 

Got errors: 

```
% make mac-app
mkdir -p bin/mac
cd server; env GOOS=darwin GOARCH=amd64 go build -ldflags '-X "github.com/mattermost/focalboard/server/model.BuildNumber=dev" -X "github.com/mattermost/focalboard/server/model.BuildDate=n/a" -X "github.com/mattermost/focalboard/server/model.BuildHash=92eb78357fbed987ef412d49667f5ea8f2ae7c47" -X "github.com/mattermost/focalboard/server/model.Edition=mac"' -o ../bin/mac/focalboard-server ./main
package github.com/mattermost/focalboard/server/main: build constraints exclude all Go files in /Users/joaoaldeano/personal/focalboard/server/main
make: *** [server-mac] Error 1
```

## Expected behavior

Looking to ReadMe this command should create an application: `Focalboard.app`.


## Edition and Platform

 - Edition: MacBook Pro (13-inch, M1, 2020) 16 GB
 - Chip: Apple M1 

## Additional context

Looking to the error it seems that Makefile line 36 defines:
`cd server; env GOOS=darwin GOARCH=amd64 go build -ldflags '$(LDFLAGS)' -o ../bin/mac/focalboard-server ./main`

But the systems arch on m1 chip is arm64.

--- a/Makefile
+++ b/Makefile
@@ -16,6 +16,13 @@ LDFLAGS += -X "github.com/mattermost/focalboard/server/model.BuildNumber=$(BUILD
 LDFLAGS += -X "github.com/mattermost/focalboard/server/model.BuildDate=$(BUILD_DATE)"
 LDFLAGS += -X "github.com/mattermost/focalboard/server/model.BuildHash=$(BUILD_HASH)"
 
+# MAC cpu architecture
+ifeq ($(shell uname -m),arm64)
+	MAC_GO_ARCH := arm64
+else
+	MAC_GO_ARCH := amd64
+endif
+
 all: webapp server ## Build server and webapp.
 
 prebuild: ## Run prebuild actions (install dependencies etc.).
@@ -33,7 +40,7 @@ server: ## Build server for local environment.
 server-mac: ## Build server for Mac.
 	mkdir -p bin/mac
 	$(eval LDFLAGS += -X "github.com/mattermost/focalboard/server/model.Edition=mac")
-	cd server; env GOOS=darwin GOARCH=$(shell uname -m) go build -ldflags '$(LDFLAGS)' -o ../bin/mac/focalboard-server ./main
+	cd server; env GOOS=darwin GOARCH=$(MAC_GO_ARCH) go build -ldflags '$(LDFLAGS)' -o ../bin/mac/focalboard-server ./main
 
 server-linux: ## Build server for Linux.
 	mkdir -p bin/linux
@@ -115,7 +122,7 @@ server-test-mysql: ## Run server tests using mysql
 	@echo Starting docker container for mysql
 	docker-compose -f ./docker-testing/docker-compose-mysql.yml run start_dependencies
 	cd server; go test -race -v -count=1 ./...
-	docker-compose -f ./docker-testing/docker-compose-mysql.yml down -v --remove-orphans 
+	docker-compose -f ./docker-testing/docker-compose-mysql.yml down -v --remove-orphans
 
 server-test-postgres: export FB_UNIT_TESTING=1
 server-test-postgres: export FB_STORE_TEST_DB_TYPE=postgres
@@ -125,7 +132,7 @@ server-test-postgres: ## Run server tests using postgres
 	@echo Starting docker container for postgres
 	docker-compose -f ./docker-testing/docker-compose-postgres.yml run start_dependencies
 	cd server; go test -race -v -count=1 ./...
-	docker-compose -f ./docker-testing/docker-compose-postgres.yml down -v --remove-orphans 
+	docker-compose -f ./docker-testing/docker-compose-postgres.yml down -v --remove-orphans
 
 webapp: ## Build webapp.
 	cd webapp; npm run pack

--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@ server: ## Build server for local environment.
 server-mac: ## Build server for Mac.
 	mkdir -p bin/mac
 	$(eval LDFLAGS += -X "github.com/mattermost/focalboard/server/model.Edition=mac")
-	cd server; env GOOS=darwin GOARCH=amd64 go build -ldflags '$(LDFLAGS)' -o ../bin/mac/focalboard-server ./main
+	cd server; env GOOS=darwin GOARCH=$(shell uname -m) go build -ldflags '$(LDFLAGS)' -o ../bin/mac/focalboard-server ./main
 
 server-linux: ## Build server for Linux.
 	mkdir -p bin/linux

